### PR TITLE
[Matrix] change to use on newest Kodi Version

### PR DESCRIPTION
**Pull request by AlwinEsch**
There are some changes to allow on Kodi Matrix, it could also maybe make sense to create a Matrix branch for them.

Also is a change related to https://github.com/xbmc/xbmc/pull/17527 included

### CODE REVIEW

**Review by afedchin — APPROVED** [review]

File: src/SMBFile.h [hunk 1]
```diff
@@ -82,8 +82,8 @@ class CSMBFile : public kodi::addon::CInstanceVFS, public P8PLATFORM::CMutex
 class ATTRIBUTE_HIDDEN CMyAddon : public kodi::addon::CAddonBase
 {
 public:
-  CMyAddon() { }
-  virtual ADDON_STATUS CreateInstance(int instanceType, std::string instanceID, KODI_HANDLE instance, KODI_HANDLE& addonInstance) override
+  CMyAddon() = default
```
Comment (AlwinEsch) [inline src/SMBFile.h]: Ups, a small typo (`;` is missing).

After the Kodi request is in I make a check of addon in Windows UWP to confirm the right operation.
EDIT: And a force for rebuild of request.

**Comment by AlwinEsch** [comment]
Is now updated and runtime tested on normal Windows and Win UWP, works correct.

Also changed to allow build on normal Windows, find there is Debugging a bit easier. Is OK for you to allow normal Windows for test purposes too?

Has also forgot before few parts of p8-platform remove, thats why two other commits with transform and chrono now there.

**Comment by AlwinEsch** [comment]
The request on Kodi is now in and AppVeyor happy :smile: 

About this:
https://github.com/xbmc/repo-binary-addons/blob/Matrix/vfs.smb2/vfs.smb2.txt#L1
Set them to master or do you create a separate Branch for Matrix?

**Comment by afedchin** [comment]
thanks

**Comment by afedchin** [comment]
Matrix branch is created

**Pull request merged**